packages/shared/pkg/storage/header/mapping.go

```diff
@@ -39,9 +38,9 @@ func CreateMapping(
 
 		if blockLength > 0 {
 			m := &BuildMap{
-				Offset:             uint64(int64(startBlock) * int64(metadata.BlockSize)),
+				Offset:             uint64(int64(startBlock) * int64(blockSize)),
 				BuildId:            *buildId,
-				Length:             uint64(blockLength) * uint64(metadata.BlockSize),
+				Length:             uint64(blockLength) * uint64(blockSize),
 				BuildStorageOffset: buildStorageOffset,
```

Comment:
Unnecessary conversion
```suggestion
				Length:             uint64(blockLength) * blockSize,
```